Security: Workflow vulnerable to bash command injection from issue/PR content

## Security Vulnerability
The GitHub Actions workflow has a command injection vulnerability when processing issue bodies, titles, and PR comments.

## Evidence
Error from workflow run #120:
```
/home/runner/work/_temp/ca4d4c21-df2c-47ad-9e97-d2216f024709.sh: line 27: productivity: command not found
Error: Process completed with exit code 127
```

This occurred when processing issue #1188 which contains "100x-1000x productivity multiplier" in the body.

## Root Cause
Direct variable assignment from GitHub Actions expressions allows bash to interpret special characters:
```bash
ISSUE_BODY="${{ github.event.issue.body }}"  # UNSAFE!
echo "$ISSUE_BODY" > /tmp/issue_body.txt     # Content gets executed!
```

If the issue body contains backticks, `$(...)`, or other shell metacharacters, they will be executed.

## Attack Vector
A malicious user could create an issue with body content like:
```
This is a test `rm -rf /` issue
```

And the workflow would attempt to execute the embedded command.

## Solution
Use heredoc syntax with single quotes to treat ALL content as literal text:
```bash
# Safe approach - no variable expansion or command substitution
cat <<'ISSUE_BODY_EOF' > /tmp/issue_body.txt
${{ github.event.issue.body }}
ISSUE_BODY_EOF
```

## Affected Lines
- Line 101: `ISSUE_TITLE="${{ github.event.issue.title }}"`
- Line 102: `ISSUE_BODY="${{ github.event.issue.body }}"` 
- Line 143: PR comment body handling

## Severity
HIGH - This allows arbitrary command execution in the GitHub Actions runner from user-controlled input.

## Fix Required
Replace all direct assignments of user input with heredoc syntax to prevent command injection.

--- a/.github/workflows/claude-implementation.yml
+++ b/.github/workflows/claude-implementation.yml
@@ -96,19 +96,27 @@ jobs:
           if [ "$IS_ISSUE" == "true" ]; then
             echo "Preparing prompt for issue implementation..."
             
-            # Get issue details
+            # Get issue number (safe - just a number)
             ISSUE_NUMBER="${{ github.event.issue.number }}"
-            ISSUE_TITLE="${{ github.event.issue.title }}"
-            ISSUE_BODY="${{ github.event.issue.body }}"
             
-            # Write all replacement values to temp files to avoid bash expansion issues
+            # Write all values to temp files to avoid bash expansion issues
+            # Use heredoc for user-controlled content to prevent command injection
             cat <<'KNOWLEDGE_EOF' > /tmp/knowledge.txt
           ${{ steps.knowledge.outputs.content }}
           KNOWLEDGE_EOF
             
             echo "$ISSUE_NUMBER" > /tmp/issue_number.txt
-            echo "$ISSUE_TITLE" > /tmp/issue_title.txt
-            echo "$ISSUE_BODY" > /tmp/issue_body.txt
+            
+            # SECURITY: Use heredoc to prevent command injection from issue title
+            cat <<'ISSUE_TITLE_EOF' > /tmp/issue_title.txt
+          ${{ github.event.issue.title }}
+          ISSUE_TITLE_EOF
+            
+            # SECURITY: Use heredoc to prevent command injection from issue body
+            cat <<'ISSUE_BODY_EOF' > /tmp/issue_body.txt
+          ${{ github.event.issue.body }}
+          ISSUE_BODY_EOF
+            
             echo "${{ github.repository }}" > /tmp/repo.txt
             
             # Load the template
@@ -150,13 +158,19 @@ jobs:
           ${{ steps.knowledge.outputs.content }}
           KNOWLEDGE_EOF
             
+            # SECURITY: Use heredoc to prevent command injection from PR comment
+            cat <<'COMMENT_BODY_EOF' > /tmp/comment_body.txt
+          ${{ github.event.comment.body }}
+          COMMENT_BODY_EOF
+            
             # Build the PR review prompt
             {
               cat /tmp/knowledge.txt
               echo ""
               echo "## PR Review Request"
               echo ""
-              echo "The user has requested: ${{ github.event.comment.body }}"
+              echo "The user has requested:"
+              cat /tmp/comment_body.txt
               echo ""
               echo "Please review and respond to this pull request comment with the full context of the codebase principles and procedures available above."
             } > /tmp/final_prompt.md

